lib/imlib/bayer.c

```diff
@@ -2291,13 +2298,19 @@ static void vdebayer_rggb_to_rgb565_awb_quarter(image_t *src, image_t *dst, uint
 // GRAYSCALE: src->data == dst->data
 // RGB565: src->data == dst->data + image_size(src)
 // YUV422: Not supported
-void imlib_debayer_image_awb(image_t *dst, image_t *src, bool fast, uint32_t r_out, uint32_t g_out, uint32_t b_out) {
+void imlib_debayer_image_awb(image_t *dst, image_t *src, bool fast, uint32_t r_out, uint32_t g_out, uint32_t b_out,
+                             float brightness, float contrast, float gamma) {
     uint32_t red_gain = IM_DIV(g_out * 32, r_out);
     red_gain = IM_MIN(red_gain, 128U);
 
     uint32_t blue_gain = IM_DIV(g_out * 32, b_out);
     blue_gain = IM_MIN(blue_gain, 128U);
 
+    for (int i = 0; i < 256; i++) {
+        int v = fast_roundf(((fast_powf(i * (1.0f / 255.0f), gamma) * contrast) + brightness) * 255.0f);
+        gamma_lut[i] = IM_CLAMP(v, 0, 255);
```

Comment:
Should this be recalculated on every call or when those values change?

Reply:
Done